airflow/hooks/hive_hooks.py

```diff
@@ -337,6 +338,26 @@ def check_for_partition(self, schema, table, partition):
         else:
             return False
 
+    def check_for_named_partition(self, schema, table, partition_name):
+        """Checks whether a partition with a given name exists
+
+        >>> hh = HiveMetastoreHook()
+        >>> t = 'static_babynames_partitioned'
+        >>> hh.check_for_named_partition('airflow', t, "ds=2015-01-01")
+        True
+        >>> hh.check_for_named_partition('airflow', t, "ds=xxx")
+        False
+        """
+        self.metastore._oprot.trans.open()
+        try:
+            partition = self.metastore.get_partition_by_name(
+                schema, table, partition_name)
+            return True
+        except hive_metastore.ttypes.NoSuchObjectException:
```

Comment:
Does this exception get thrown if the table is missing as well, or only if the partition is missing?